R/zchunk_L2233.electricity_water.R

```diff
@@ -84,406 +110,888 @@ module_water_L2233.electricity_water_DISABLED <- function(command, ...) {
+      add_precursors("L201.en_bcoc_emissions",
+                     "temp-data-inject/L223.StubTechEff_elec",
+                     "temp-data-inject/L241.nonco2_tech_coeff") ->
       L2233.InputEmissCoeff_hist_elecPassthru
 
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L2233.InputEmissCoeff_fut_elecPassthru %>%
+      add_title("BC and OC emissions coefficients from electric power plants (future)") %>%
+      add_units("TBC") %>%
+      add_comments("Emissions coefficients corrected by diving by elec tech efficiencies") %>%
       add_legacy_name("L2233.InputEmissCoeff_fut_elecPassthru") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("temp-data-inject/L223.GlobalTechEff_elec",
+                     "temp-data-inject/L241.nonco2_tech_coeff") ->
       L2233.InputEmissCoeff_fut_elecPassthru
 
-    return_data(object, new_object, L2233.SectorNodeEquiv, L2233.TechNodeEquiv, L2233.StubTech_elecPassthru, L2233.StubTechProd_elecPassthru, L2233.GlobalPassThroughTech, L2233.GlobalTechEff_elecPassthru, L2233.GlobalTechShrwt_elecPassthru, L2233.GlobalIntTechCapital_elec, L2233.GlobalTechCapital_elecPassthru, L2233.GlobalIntTechOMfixed_elec, L2233.GlobalTechOMfixed_elecPassthru, L2233.GlobalIntTechOMvar_elec, L2233.GlobalTechOMvar_elecPassthru, L2233.PassThroughSector_elec_cool, L2233.SectorLogitTables_elec_cool[[ curr_table ]]$data, L2233.Supplysector_elec_cool, L2233.ElecReserve_elec_cool, L2233.SubsectorShrwtFllt_elec_cool, L2233.SubsectorLogitTables_elec_cooling[[ curr_table ]]$data, L2233.SubsectorLogit_elec_cool, L2233.StubTech_elec_cool, L2233.StubTechEff_elec_cool, L2233.StubTechProd_elec_cool, L2233.StubTechFixOut_hydro, L2233.StubTechShrwt_elec_cool, L2233.GlobalTechCapital_elec_cool, L2233.GlobalIntTechCapital_elec_cool, L2233.GlobalTechCoef_elec_cool, L2233.GlobalIntTechCoef_elec_cool, L2233.InputEmissCoeff_hist_elecPassthru, L2233.InputEmissCoeff_fut_elecPassthru)
+    ## LAST TWO OUTPUTS ARE SIMPLY TAG NAMES (USED TO AVOID HAVING TO PARTITION INPUT TABLES)
+    tibble(X1 = "SectorXMLTags", X2 = "supplysector", X3 = "pass-through-sector" ) %>%
+      add_title("Equivalent sector tag names") %>%
+      add_units("NA") %>%
+      add_comments("") %>%
+      add_legacy_name("L2233.SectorNodeEquiv")  ->
+      L2233.SectorNodeEquiv
+
+    tibble(X1 = "TechnologyXMLTags", X2 = "technology", X3 = "intermittent-technology", X4 = "pass-through-technology") %>%
+      add_title("Equivalent technology tag names") %>%
+      add_units("NA") %>%
+      add_comments("") %>%
+      add_legacy_name("L2233.TechNodeEquiv") ->
+      L2233.TechNodeEquiv
+
+    return_data(L2233.SectorNodeEquiv,
```

Comment:
We eventually consolidate all of these files into one or two XML files for the actual runs. We want to replicate what is there for now, but we should really consider consolidating all of these files and taking better advantage of R/DSR capability. This R chunk could be significantly condensed eventually.